### PR TITLE
Add offline capabilities to the shadow OVAL probe

### PR DESCRIPTION
**Pull request by 0intro**
This change adds offline capabilities to the shadow probe.

### CODE REVIEW

File: src/OVAL/probes/unix/shadow_probe.c [hunk 1]
```diff
@@ -177,36 +177,64 @@ static void report_finding(struct result_info *res, probe_ctx *ctx)
...
-			r.chg_req = pw->sp_max;
-			r.exp_warn = pw->sp_warn;
-			r.exp_inact = pw->sp_inact;
-			r.exp_date = pw->sp_expire;
-			r.flag = pw->sp_flag;
-
-			report_finding(&r, ctx);
+	if (ctx->offline_mode & PROBE_OFFLINE_OWN) {
+		const char *root = getenv("OSCAP_PROBE_ROOT");
+		if (root == NULL)
```
Comment (evgenyz) [inline src/OVAL/probes/unix/shadow_probe.c:212]: The `oscap_path_join` function can handle NULL prefix and presence of `OSCAP_PROBE_ROOT` variable is the trigger for offline mode. I'd say this check is redundant.

**Review by evgenyz — APPROVED** [review]
All in all — LGTM. Thanks.

File: src/OVAL/probes/unix/shadow_probe.c [hunk 2]
```diff
@@ -177,36 +177,64 @@
...
+			return 1;
+		char *shadow_file_path = oscap_path_join(root, "/etc/shadow");
+		FILE *fp = fopen(shadow_file_path, "r");
```
[the remark this answers is not in the document]
Comment (evgenyz) [reply src/OVAL/probes/unix/shadow_probe.c:215]: Yeah, yeah.

File: src/OVAL/probes/unix/shadow_probe.c (hunk 1 above)
Comment (0intro) [reply src/OVAL/probes/unix/shadow_probe.c]: Fixed. Thanks.

**Pull request merged**